uniffi_bindgen/src/bindings/kotlin/gen_kotlin/error.rs

```diff
@@ -2,89 +2,48 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-use std::fmt;
-
-use crate::bindings::backend::{CodeDeclaration, CodeOracle, CodeType, Literal};
-use crate::interface::{ComponentInterface, Error};
+use super::{names, CodeBuilder, KotlinCodeName, KotlinCodeType};
+use crate::interface::types::ErrorTypeHandler;
+use crate::interface::{ComponentInterface, Error, Variant};
 use askama::Template;
 
-use super::filters;
-pub struct ErrorCodeType {
-    id: String,
-}
-
-impl ErrorCodeType {
-    pub fn new(id: String) -> Self {
-        Self { id }
-    }
-}
-
-impl CodeType for ErrorCodeType {
-    fn type_label(&self, oracle: &dyn CodeOracle) -> String {
-        oracle.class_name(&self.id)
-    }
-
-    fn canonical_name(&self, oracle: &dyn CodeOracle) -> String {
-        format!("Error{}", self.type_label(oracle))
-    }
-
-    fn literal(&self, _oracle: &dyn CodeOracle, _literal: &Literal) -> String {
-        unreachable!();
-    }
-
-    fn lower(&self, oracle: &dyn CodeOracle, nm: &dyn fmt::Display) -> String {
-        format!("{}.lower()", oracle.var_name(nm))
+impl KotlinCodeType for ErrorTypeHandler<'_> {
+    fn nm(&self) -> String {
+        names::error_name(self.name)
     }
 
-    fn write(
-        &self,
-        oracle: &dyn CodeOracle,
-        nm: &dyn fmt::Display,
-        target: &dyn fmt::Display,
-    ) -> String {
-        format!("{}.write({})", oracle.var_name(nm), target)
-    }
-
-    fn lift(&self, oracle: &dyn CodeOracle, nm: &dyn fmt::Display) -> String {
-        format!("{}.lift({})", self.type_label(oracle), nm)
+    fn declare_code(&self, code_builder: CodeBuilder, ci: &ComponentInterface) -> CodeBuilder {
+        code_builder.code_block(KotlinError::new(
+            ci.get_error_definition(self.name)
+                .expect("Error definition not found")
+                .clone(),
```

Comment:
How hard would it be to store a reference in `KotlinError` rather than doing a `clone` here? Naively it seems tractable, because the reference `ci` should comfortably outlive the entire rendering process.